Backend/ManagerLayer/Login/UserLoginManager.cs

```diff
@@ -0,0 +1,97 @@
+﻿using DataAccessLayer.Database;
+using DataAccessLayer.Models;
+using ManagerLayer.AccessControl;
+using ManagerLayer.Models;
+using ManagerLayer.UserManagement;
+using ServiceLayer.Services;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace ManagerLayer.Login
+{
+    public class UserLoginManager
+    {
+        IPasswordService _passwordService;
+        UserManagementManager _userManagementManager;
+        AuthorizationManager _authorizationManager;
+
+        public LoginManagerResponsePOCO Login(string username, string password)
+        {
+            LoginManagerResponsePOCO response;
+            _userManagementManager = new UserManagementManager();
+            _passwordService = new PasswordService();
+            var user = _userManagementManager.GetUser(username);
+            if (user == null)
+            {
+                response = new LoginManagerResponsePOCO
+                {
+                    Data = null,
+                    Message = "User does not exist",
+                    Timestamp = DateTime.UtcNow
+                };
+                return response;
+            }
+            var StoredPassword = user.PasswordHash;
+            var AttemptedPassword = _passwordService.HashPassword(password, user.PasswordSalt);
+            if (StoredPassword == AttemptedPassword)
+            {
+                string token = _authorizationManager.CreateSession(user);
+                return new LoginManagerResponsePOCO
+                {
+                    Data = token,
+                    Message = "Login Successful",
+                    Timestamp = DateTime.UtcNow
+                };
+            }
+            return new LoginManagerResponsePOCO
+            {
+                Data = null,
+                Message = "Invalid password entered",
+                Timestamp = DateTime.UtcNow
+            };
+        }
+
+        public LoginManagerResponsePOCO LoginFromSSO(string username, string ssoID)
+        {
+            LoginManagerResponsePOCO response;
+            _userManagementManager = new UserManagementManager();
+            _authorizationManager = new AuthorizationManager();
+            var user = _userManagementManager.GetUser(username);

```

Comment:
Don't search by username (email) - it will change. Search by SSOUserId, which will be persisted (static) between every launch.

With the way you have it now, if a user changes their email address in the SSO, the next time they launch they will be a new user within PointMap.

Reply:
Same question as earlier regarding SSOUserId